### ISSUE TITLE
How to test Actor that uses DependencyResolver to create child?

### ISSUE THREAD

**Issue by tjaart** [AkkaNetContrib/Akka.TestKit.NUnit#40]
Hi there. 

I am experimenting with Akka .Net for a PoC, and tried to write some tests with the NUnit TestKit. However I am unable to test my Actor that uses the DependencyResolver.

If an actor is using dependency resolver internally:
```csharp
var actorProps = DependencyResolver.For(Context.System).Props<MyTestActor>();
var actorRef = Context.System.ActorOf(actorProps , "my-test-actor-name");
```

It doesn't seem possible to set up this actor on TestKit for NUnit, because NUnit is throwing an exception when being supplied with the ActorSystem instance:

```csharp
public TestKit(ActorSystem system = null) : base((ITestKitAssertions) Akka.TestKit.NUnit.TestKit._assertions, system)
{
    if (system != null)
        throw new NotSupportedException("Due to the way NUnit works, providing an ActorSystem is not supported.  For further details please see https://github.com/akkadotnet/akka.net/pull/1092");
}
```

I am unsure if I am using the DependencyResolver incorrectly by calling it directly in my actor, or if NUnit TestKit does not support this use case?

**Comment by xas** [comment]
We had the same problem and we were finally able to make it work with ServiceCollection build running before the test class instanciation.
You have to provide your own ActorSystem to the TestKit class instead of its own implementation.

So basically we have something like 
```csharp
var serviceCollection = new ServiceCollection();
serviceCollection.AddTransient(Mock<...>);
...
serviceCollection.AddAkka("testing", builder =>
{
    builder.AddHocon(ConfigurationFactory.ParseString(the_default__hocon_config));
}
var provider = serviceCollection.BuildServiceProvider();
...
var actorSystem = provider.GetService<ActorSystem>();
```

and in the test class we were able to have

```csharp
public class MyTest : TestKit
{
    protected override void AfterAll() { }

    public MyTest(ActorSystem actorSystem) : base(actorSystem) { }
}
```

We used xunit and the `IClassFixture` but should be possible in nunit with the `SetUpFixture` attribute

_Remarks_ :

* The `AfterAll()` override was needed as we stumble on some problem during the cleaning after each tests, we have to investigate more about it
* _the_default__hocon_config_ is from [here](https://getakka.net/articles/configuration/modules/akka.testkit.html)

**Comment by Aaronontheweb** [comment]
FWIW @xas - we're considering shipping a new TestKit-ish package built on top of Akka.Hosting to make it easier to use the `IServiceProvider` and `IHost` inside tests. We've found it to be much easier to write larger integration tests this way but it's been a "roll your own" experience even for us so far.

**Comment by Unicodist** [comment]
@Aaronontheweb Any updates on the package?

**Comment by Aaronontheweb** [comment]
Akka.Hosting.TestKit was released as part of 1.5 but we only support xUnit - but that's the proper way to do this type of testing. Guess we can close this.
